modules/flowable-engine-common/src/main/java/org/flowable/common/engine/impl/javax/el/Util.java

```diff
@@ -0,0 +1,741 @@
+                    boolean b1 = Number.class.isAssignableFrom(type1) || type1.isPrimitive();
+                    boolean b2 = Number.class.isAssignableFrom(type2) || type2.isPrimitive();
+                    if (b1 && !b2) {
+                        return 1;
+                    } else if (b2 && !b1) {
+                        return -1;
+                    } else {
+                        return 0;
+                    }
+                }
+
+                return 0;
+            } else {
+                return 0;
+            }
+        }
+    }
+
+    /*
+     * This method duplicates code in com.sun.el.util.ReflectionUtil. When making changes keep the code in sync.
+     */
+    private static Class<?> getBoxingTypeIfPrimitive(Class<?> clazz) {
+        if (clazz.isPrimitive()) {
+            if (clazz == Boolean.TYPE) {
+                return Boolean.class;
+            }
+            if (clazz == Character.TYPE) {
+                return Character.class;
+            }
+            if (clazz == Byte.TYPE) {
+                return Byte.class;
+            }
+            if (clazz == Short.TYPE) {
+                return Short.class;
+            }
+            if (clazz == Integer.TYPE) {
+                return Integer.class;
+            }
+            if (clazz == Long.TYPE) {
+                return Long.class;
+            }
+            if (clazz == Float.TYPE) {
+                return Float.class;
+            }
+
+            return Double.class;
+        } else {
+            return clazz;
+        }
+    }
+
+    /*
+     * This method duplicates code in com.sun.el.util.ReflectionUtil. When making changes keep the code in sync.
+     */
+    private static Class<?>[] getComparingParamTypesForVarArgsMethod(Class<?>[] paramTypes, int length) {
+        Class<?>[] result = new Class<?>[length];
+        System.arraycopy(paramTypes, 0, result, 0, paramTypes.length - 1);
+        Class<?> type = paramTypes[paramTypes.length - 1].getComponentType();
+        for (int i = paramTypes.length - 1; i < length; i++) {
+            result[i] = type;
+        }
+
+        return result;
+    }
+
+
+    private static final String paramString(Class<?>[] types) {
+        if (types != null) {
+            StringBuilder sb = new StringBuilder();
+            for (Class<?> type : types) {
+                if (type == null) {
+                    sb.append("null, ");
+                } else {
+                    sb.append(type.getName()).append(", ");
+                }
+            }
+            if (sb.length() > 2) {
+                sb.setLength(sb.length() - 2);
+            }
+            return sb.toString();
+        }
+        return null;
+    }
+
+
+    /*
+     * This method duplicates code in org.apache.el.util.ReflectionUtil. When
+     * making changes keep the code in sync.
+     */
+    private static <T> Wrapper<T> resolveAmbiguousWrapper(Set<Wrapper<T>> candidates,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.